SIGSEGV on APIgateway log group ls

Marks-MacBook-Pro[13:32:55] cw --profile=zjdevacct --region=us-east-2 ls streams "API-Gateway-Execution-Logs_72oi6rv36b/Alpha"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13d6c8b]

goroutine 8 [running]:
github.com/lucagrulla/cw/cloudwatch.(*CW).LsStreams.func1.1(0x1, 0x0, 0xc0000c3db8)
	/Users/lucagrulla/Projects/cw/cloudwatch/lsstreams.go:24 +0x2b
sort.insertionSort_func(0xc0000c3e38, 0xc000487c00, 0x0, 0x14)
	/usr/local/Cellar/go/1.11.5/libexec/src/sort/zfuncversion.go:12 +0x93
sort.stable_func(0xc0000c3e38, 0xc000487c00, 0x32)
	/usr/local/Cellar/go/1.11.5/libexec/src/sort/zfuncversion.go:167 +0x51
sort.SliceStable(0x146ace0, 0xc000487be0, 0xc0000c3e38)
	/usr/local/Cellar/go/1.11.5/libexec/src/sort/slice.go:31 +0x106
github.com/lucagrulla/cw/cloudwatch.(*CW).LsStreams.func1(0xc00000b460, 0xc000030a00, 0xc0000556b8)
	/Users/lucagrulla/Projects/cw/cloudwatch/lsstreams.go:23 +0x81
github.com/aws/aws-sdk-go/service/cloudwatchlogs.(*CloudWatchLogs).DescribeLogStreamsPagesWithContext(0xc00000c270, 0x1a74188, 0xc0000260b0, 0xc000183320, 0xc000061820, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/lucagrulla/go/pkg/mod/github.com/aws/aws-sdk-go@v1.14.17/service/cloudwatchlogs/api.go:1670 +0x116
github.com/aws/aws-sdk-go/service/cloudwatchlogs.(*CloudWatchLogs).DescribeLogStreamsPages(0xc00000c270, 0xc000183320, 0xc000061820, 0x0, 0x0)
	/Users/lucagrulla/go/pkg/mod/github.com/aws/aws-sdk-go@v1.14.17/service/cloudwatchlogs/api.go:1642 +0x90
github.com/lucagrulla/cw/cloudwatch.(*CW).LsStreams.func2(0xc000061810, 0xc000183320, 0xc000061820)
	/Users/lucagrulla/Projects/cw/cloudwatch/lsstreams.go:37 +0x49
created by github.com/lucagrulla/cw/cloudwatch.(*CW).LsStreams
	/Users/lucagrulla/Projects/cw/cloudwatch/lsstreams.go:36 +0xf1

Only the API-Gateway log group is throwing the panic.

Marks-MacBook-Pro[12:25:05] awsdev --query logGroups[*].logGroupName  logs describe-log-groups
[
    "/aws/lambda/shipper-sam-ShipperHandlerFunction-155P3N3JNQEE4",
    "/aws/rds/instance/zj-dev-postgres/postgresql",
    **"API-Gateway-Execution-Logs_joekpkdl9f/Alpha",**
    "RDSOSMetrics"
]

===== API-Gateway =====

Marks-MacBook-Pro[12:27:45] cw --profile=zjdevacct --region=us-east-2 ls streams "API-Gateway-Execution-Logs_joekpkdl9f/Alpha"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13d6c8b]

goroutine 21 [running]:
github.com/lucagrulla/cw/cloudwatch.(*CW).LsStreams.func1.1(0x1, 0x0, 0xc0000d5db8)
	/Users/lucagrulla/Projects/cw/cloudwatch/lsstreams.go:24 +0x2b
sort.insertionSort_func(0xc0000d5e38, 0xc000379b40, 0x0, 0x14)
	/usr/local/Cellar/go/1.11.5/libexec/src/sort/zfuncversion.go:12 +0x93
sort.stable_func(0xc0000d5e38, 0xc000379b40, 0x32)
	/usr/local/Cellar/go/1.11.5/libexec/src/sort/zfuncversion.go:167 +0x51
sort.SliceStable(0x146ace0, 0xc000379b20, 0xc0000d5e38)
	/usr/local/Cellar/go/1.11.5/libexec/src/sort/slice.go:31 +0x106
github.com/lucagrulla/cw/cloudwatch.(*CW).LsStreams.func1(0xc0000a5400, 0xc000094900, 0xc00004feb8)
	/Users/lucagrulla/Projects/cw/cloudwatch/lsstreams.go:23 +0x81
github.com/aws/aws-sdk-go/service/cloudwatchlogs.(*CloudWatchLogs).DescribeLogStreamsPagesWithContext(0xc00009e260, 0x1a782a0, 0xc000092020, 0xc00019b320, 0xc000081810, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/lucagrulla/go/pkg/mod/github.com/aws/aws-sdk-go@v1.14.17/service/cloudwatchlogs/api.go:1670 +0x116
github.com/aws/aws-sdk-go/service/cloudwatchlogs.(*CloudWatchLogs).DescribeLogStreamsPages(0xc00009e260, 0xc00019b320, 0xc000081810, 0x0, 0x0)
	/Users/lucagrulla/go/pkg/mod/github.com/aws/aws-sdk-go@v1.14.17/service/cloudwatchlogs/api.go:1642 +0x90
github.com/lucagrulla/cw/cloudwatch.(*CW).LsStreams.func2(0xc000081800, 0xc00019b320, 0xc000081810)
	/Users/lucagrulla/Projects/cw/cloudwatch/lsstreams.go:37 +0x49
created by github.com/lucagrulla/cw/cloudwatch.(*CW).LsStreams
	/Users/lucagrulla/Projects/cw/cloudwatch/lsstreams.go:36 +0xf1

===== RDS log group =====

Marks-MacBook-Pro[12:28:57] cw --profile=zjdevacct --region=us-east-2 ls streams "/aws/rds/instance/zj-dev-postgres/postgresql"
zj-dev-postgres.0

===== Lambda log group =====

Marks-MacBook-Pro[12:29:49] cw --profile=zjdevacct --region=us-east-2 ls streams "/aws/lambda/shipper-sam-ShipperHandlerFunction-155P3N3JNQEE4"
2019/04/22/[$LATEST]be0bbbbfae6d473fadb3522c341a24aa
2019/04/22/[$LATEST]067580023a4c4ff5909adf5246b20b05
2019/04/22/[$LATEST]93d395598ffa4d3f87d49f2e1cef576b
~/Developer/ZoneJumpDev/GitHub/zj-shipper-api
Marks-MacBook-Pro[12:30:19]

--- a/cloudwatch/lsstreams.go
+++ b/cloudwatch/lsstreams.go
@@ -21,7 +21,18 @@ func (cwl *CW) LsStreams(groupName *string, streamName *string) <-chan *string {
 	}
 	handler := func(res *cloudwatchlogs.DescribeLogStreamsOutput, lastPage bool) bool {
 		sort.SliceStable(res.LogStreams, func(i, j int) bool {
-			return *res.LogStreams[i].LastIngestionTime < *res.LogStreams[j].LastIngestionTime
+			var streamALastIngestionTime int64 = 0;
+			var streamBLastIngestionTime int64 = 0;
+
+			if ingestionTime := res.LogStreams[i].LastIngestionTime; ingestionTime != nil {
+				streamALastIngestionTime = *ingestionTime;
+			}
+
+			if ingestionTime := res.LogStreams[j].LastIngestionTime; ingestionTime != nil {
+				streamBLastIngestionTime = *ingestionTime;
+			}
+
+			return streamALastIngestionTime < streamBLastIngestionTime;
 		})
 
 		for _, logStream := range res.LogStreams {

